### ISSUE TITLE
Bug in channel creation with permissions

### ISSUE THREAD

**Issue by kzay** [hydrabolt/discord.js#2263]
**Please describe the problem you are having in as much detail as possible:**

There is a bug with channel creation with permissions.

In case of RoleId this code never resolve the role and go for type=member

   => const role = this.client.resolver.resolveRole(this, overwrite.id);

I think this must be replaced with guild.

**Include a reproducible code sample here, if possible:**
```js

  createChannel(guild, channelName, channelType, overwrites, reason) {
    if (overwrites instanceof Collection || overwrites instanceof Array) {
      overwrites = overwrites.map(overwrite => {
        let allow = overwrite.allow || overwrite._allowed;
        let deny = overwrite.deny || overwrite._denied;
        if (allow instanceof Array) allow = Permissions.resolve(allow);
        if (deny instanceof Array) deny = Permissions.resolve(deny);

        const role = this.client.resolver.resolveRole(this, overwrite.id);
        if (role) {
          overwrite.id = role.id;
          overwrite.type = 'role';
        } else {
          overwrite.id = this.client.resolver.resolveUserID(overwrite.id);
          overwrite.type = 'member';
        }

        return {
          allow,
          deny,
          type: overwrite.type,
          id: overwrite.id,
        };
      });
    }
    return this.rest.makeRequest('post', Endpoints.Guild(guild).channels, true, {
      name: channelName,
      type: Constants.ChannelTypes[channelType.toUpperCase()],
      permission_overwrites: overwrites,
    }, undefined, reason).then(data => this.client.actions.ChannelCreate.handle(data).channel);
  }

```

**Further details:**

- discord.js version: 11.3
- node.js version:
- Operating system:
- Priority this issue should have – please be realistic and elaborate if possible: Major

- [ ] I found this issue while running code on a __user account__
- [ ] I have also tested the issue on latest master, commit hash:

**Comment by Lewdcario** [comment]
Can you post a proper reproducible code sample, that which does not come from the library?

**Comment by SpaceEEC** [comment]
I did a bit of debugging:

Something like this would serve as example:
`guild.createChannel('someName', 'text', [{ deny: 1024, id: guild.id }]);`

This is not applying the expected overwrite (denying `@everyone` to read messages)
because resolving of the role fails, if it's a role id.

Which happens because we pass `this` (RestManager) as guild to `ClientDataResolver#resolveRole`, opposed to the available variable `guild` [here](https://github.com/hydrabolt/discord.js/blob/stable/src/client/rest/RESTMethods.js#L263).

A workaround for now would be to pass the role object itself instead of the role id.

**Comment by kzay** [comment]
This is true, i'm using this workqround but there is a limit to it.

If you are using a variable for the permission list like this.

var permission = [{ deny: 1024, id: someRoleObject }];
guild.createChannel('someName', 'text', permission );
guild.createChannel('someName2', 'text', permission );
guild.createChannel('someName3', 'text', permission );

At the first creation the permission will work, but the array will be modified by reference here : 

overwrites = overwrites.map(overwrite => {

The someRoleObject will be replaced by the roleId. And for the second creation the bug will occure and the array will be modified again with the type : member. So all the other creation after the first one wont work.

**Comment by Lewdcario** [comment]
Resolved in the dev branch `11.3.1-dev`, as far as testing it goes.
